carto/auth.py

```diff
@@ -0,0 +1,106 @@
+"""
+Module for authenticated access to CARTO's APIs
+
+.. module:: carto.auth
+   :platform: Unix, Windows
+   :synopsis: Module for authenticated access to CARTO's APIs
+
+.. moduleauthor:: Daniel Carrion <daniel@carto.com>
+.. moduleauthor:: Alberto Romeu <alrocar@carto.com>
+
+
+"""
+
+import re
+import sys
+import warnings
+
+from pyrestcli.auth import BaseAuthClient
+
+from .exceptions import CartoException
+
+if sys.version_info >= (3, 0):
```

Comment:
Maybe `six` is handier, but since we are doing minor checks, let's keep with `sys` to avoid adding another dependency